### ISSUE TITLE
Simple sample fails.

### ISSUE THREAD

**Issue by pward84020** [google/closure-compiler#2760]
Anyone know why this simple snippet (kinda like a static class) - which runs perfectly fine....

var classA =
    {
      fname : "Peter",
      print : function() {alert(this.fname); }
    };
classA.print(); 

Produces this incorrect compiled code ? (ADVANCED_OPTIMIZATIONS)

(function(){alert(this.a)})();

Thanks,

**Comment by MatrixFrog** [comment]
https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%250Avar%2520classA%2520%253D%2520%257B%250A%2520%2520fname%253A%2520%2522Peter%2522%252C%250A%2520%2520print%253A%2520function()%2520%257Balert(this.fname)%253B%2520%257D%252C%250A%257D%253B%250AclassA.print()%253B

**Comment by ChadKillingsworth** [comment]
That's definitely an unsafe `this` reference. CollapseProperties should be warning in advanced mode.

```js
var classA =
{
fname : "Peter",
print : function() {alert(this.fname); } // should be classA.fname
};
classA.print();
```

**Comment by ChadKillingsworth** [comment]
@MatrixFrog Your example is still in ADVANCED mode. Simple looks correct to me.

**Comment by MatrixFrog** [comment]
So this is WAI then, right? It might be helpful if the unsafe this warning said something like "Property collapsing may transform correct code into incorrect code" though.

**Comment by ChadKillingsworth** [comment]
Yes - it's working as intended. And the warning most certainly could be improved.

**Comment by pward84020** [comment]
I find it confusing that the non-compiled code produces an ALERT POPUP with 'Peter' in it.
And the compiled code produces an ALERT POPUP with 'undefined' in it.

Is there a way to get this to compile and work ?

Should a dangerous use of 'this' produce such erroneous code?
JSC_UNSAFE_THIS: dangerous use of 'this' in static method classA.print at line 3 character 27

Unfortunately, we use this type of 'static class' quite a bit in our code.  Are there any annotations which I can tag onto it ?

Thanks again.

**Comment by MatrixFrog** [comment]
For the moment, you can use simple mode, or Chad's suggestion. We have discussed a safer version of CollapseProperties but I can't promise anything in the near future.

**Comment by pward84020** [comment]
Okay :(
Does the code get compiled in SIMPLE MODE ?
ie, Can it detect syntax errors, undeclared variables, code stripping ?  Any of the features I wanted from ADVANCED MODE ?   I'm looking for some degree of error checking besides errors occurring while running.

**Comment by MatrixFrog** [comment]
Error checking is about the same in simple mode, you just get less advanced optimizations.
